app/template_folder/rest.py

```diff
@@ -89,3 +92,57 @@ def delete_template_folder(service_id, template_folder_id):
     dao_delete_template_folder(template_folder)
 
     return '', 204
+
+
+@template_folder_blueprint.route('/move-to-folder', methods=['POST'])
+@template_folder_blueprint.route('/move-to-folder/<uuid:target_template_folder_id>', methods=['POST'])
```

Comment:
It doesn't solve everything, but I think this would be a bit more RESTful if we think of this as updating folder contents instead of a "move" action. So instead of `/move-to-folder/<uuid>` we'd have `POST /template-folder/<uuid>/contents`. This is still a bit awkward without any ID for "top-level", but I think `/template-folder/contents` could work.

One of the benefits is that we can add a `GET` endpoint to retrieve everything in the folder if we need it in the future and model this as a proper resource instead of having a number of unconnected remote procedures.